### PR TITLE
+ actor #15446 add external predicate to ReceiveBuilder

### PR DESCRIPTION
**Pull request by He-Pin**
refs https://github.com/akka/akka/issues/15446

### CODE REVIEW

**Comment by hepin1989** [comment]
tests is not added for now.

File: akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java [hunk 1]
```diff
@@ -130,6 +130,23 @@ public boolean defined(Object o) {
...
+   * @param type
+   *          a type to match the argument against
+   * @param externalPredicate
+   *          a external predicate that will be evaluated if the type matches
+   * @param apply
+   *          an action to apply to the argument if the type matches and the
+   *          predicate returns true
+   * @return a builder with the case statement added
+   */
+  public <P> ReceiveBuilder match(final Class<P> type, final java.util.function.BooleanSupplier externalPredicate,
```
Comment (ktoso) [inline akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java]: Not sure this really is worth it. We already have `FI.TypedPredicate` which you can simply use as `_ -> false` done

Comment (hepin1989) [reply akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java]: it's original your issue :)

Comment (ktoso) [reply akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java]: lol :) 
I guess it'd be ok to add, answered without reading the ticket... :)

**Review by johanandren — CHANGES_REQUESTED** [review]
Missing test coverage

**Review by johanandren — APPROVED** [review]

**Comment by hepin1989** [comment]
rebased

**Review by ktoso — APPROVED** [review]
LGTM

**Pull request merged**